.github/workflows/tests.yml

```diff
@@ -27,13 +27,74 @@ jobs:
       - name: Rebase to current master
         run: git rebase origin/master
 
+      - name: Prepare
+        run: |
+          # # override GitHub's bind mount from host, we don't want anything from there and it interferes with ssh
+          # export HOME=$(getent passwd $(id -u) | cut -f6 -d:)
+
+          # Put the /secrets in place which we'll mount to the tasks-container
+          mkdir -p ~/secrets
+          echo '${{ secrets.COCKPITUOUS_TOKEN }}' > ~/.config/github-token
+          echo '${{ secrets.SSH_KNOWN_HOSTS }}' > ~/secrets/known_hosts
+          echo '${{ secrets.SINK_SSH_CONFIG }}' > ~/secrets/ssh-config
+          echo '${{ secrets.SINK_ID_RSA }}' > ~/secrets/id_rsa
+          echo '${{ secrets.SINK_ID_RSA_PUB }}' > ~/secrets/id_rsa.pub
+          sudo chmod 400 ~/secrets/*
+          # change permissions to user in cockpit-tasks
+          # TODO sudo needed?
+          sudo chown 1111:1111 ~/secrets/*
+
+          # install dependencies needed to use bots stuff in the host container
+          sudo pip3 install pika
+
+      # - name: Setup tmate session
+      #   uses: mxschmitt/action-tmate@v3
+      #   timeout-minutes: 45
+
       - name: Build and test tasks container if it changed
         run: |
           changes=$(git diff --name-only origin/master..HEAD -- tasks/)
           if [ -n "${changes}" ]; then
               # Run podman as root, as podman is missing slirp4netns by default, and does not have overlayfs by default
-              sudo make tasks-container
-              # This does not do much yet, just iterate to the AMQP queue 30 times and exit
-              # TODO: Inject a workload that does not need /dev/kvm
-              sudo tasks/run-local.sh
+              # TODO uncomment, make it faster now
+              # sudo make tasks-container
+
+              # Generate the secrets in tree and copy the working github token
+              (cd tasks/credentials; ./generate-ca.sh)
+              (cd tasks/credentials/webhook; ./generate.sh)
+              chmod -R go+rX tasks/credentials/webhook
+              cp ~/.config/github-token tasks/credentials/webhook/.config--github-token
+
+              # Start the amqp/tasks pod in the background and supply working task secrets
+              sudo tasks/run-local.sh -d -s ~/secrets
+
+              # clone bots
+              # TODO: figure out how to split this into different steps if possible
+              git clone https://github.com/cockpit-project/bots
+              git -C ./bots fetch origin pull/1567/head
+              git -C ./bots checkout FETCH_HEAD
```

Comment:
The tasks container already clones the bots -- I really think it's cleaner and also more useful (as it validates much more actual code) to do this inside the cockpituous-tasks container.

Reply:
yea this was just because I needed to check out the specific bots PR so it would recognize the test it triggers in the testmap.